atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameMapperTest.java

```diff
@@ -87,6 +89,20 @@ public void shouldThrowIfTable99999Exists() {
         oracleTableNameMapper.getShortPrefixedTableName(connectionSupplier, TEST_PREFIX, tableRef);
     }
 
+    @Test
+    public void shouldGetRightPrimaryKeyConstraintForLongTableNames() {
+        String pkConstraintName = OraclePrimaryKeyConstraintNames.get(LONG_TABLE_NAME);
+        assertThat(pkConstraintName, is(StringUtils.left("pk_" + LONG_TABLE_NAME, 30)));
+        assertThat(pkConstraintName.length(), is(OracleTableNameMapper.ORACLE_MAX_TABLE_NAME_LENGTH));
```

Comment:
what's the value of the constraint? We probably want to test behavior of long values that would truncate to the same values:

```
assertThat(OraclePrimaryKeyConstraintNames.get("abcdefghijklmnopqrstuvwxyz1234567890A"),
is(not(equalTo(OraclePrimaryKeyConstraintNames.get("abcdefghijklmnopqrstuvwxyz1234567890B"))));
```